### DISCUSSION TITLE
Question about Entropy Coder precision

### DISCUSSION THREAD

**Discussion by aguarda** [tensorflow/compression#115] [Q&A]
Hello everyone,

I noticed that in TFC v1.3 the default entropy coder precision was set to 16 bits, but as of the current v2.2, the default entropy coder precision is 12 bits. Is there a particular reason for this change? What is the impact on the performance?

Thank you!

**Comment by jonaballe** [comment]
Hi André, there is no particular reason for it. We haven't done extensive testing on what the impact is on performance with this implementation. Obviously, the higher the precision, the closer the rate is to the Shannon entropy, but the higher is also the computational complexity. There is a [unit test that checks the overhead](https://github.com/tensorflow/compression/blob/master/tensorflow_compression/python/entropy_models/continuous_batched_test.py#L145), but of course this can differ with the data distribution.
